Unify copies of WebKitTargetConditionals.xcconfig

ddkilzer opened the pull request:
#### 3f8075f50078abbf0feeb5f907b402139f28b2d9
<pre>
Unify copies of WebKitTargetConditionals.xcconfig
<a href="https://bugs.webkit.org/show_bug.cgi?id=244440">https://bugs.webkit.org/show_bug.cgi?id=244440</a>
&lt;rdar://99233826&gt;

Reviewed by Alexey Proskuryakov.

- Move latest copy of WebKitTargetConditionals.xcconfig into
  Configurations folder.
- Remove duplicate copies of WebKitTargetConditionals.xcconfig
  in each project.
- Remove WebKitTargetConditionals.xcconfig from Xcode projects.
- Change references to WebKitTargetConditionals.xcconfig in
  other xcconfig files to use a relative path to the single
  copy.

* Configurations/WebKitTargetConditionals.xcconfig: Rename from Source/WebGPU/Configurations/WebKitTargetConditionals.xcconfig.
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/AngleMetalLib.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebInspectorUI/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
- Remove import of WebKitTargetConditionals.xcconfig since it&apos;s
  already included in BaseTarget.xcconfig.
* Source/WebKit/Configurations/WebKitPlatform.xcconfig: Ditto.
* Source/WebKit/Configurations/WebKitSwift.xcconfig: Ditto.
* Source/WebKit/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/253873@main">https://commits.webkit.org/253873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/488609a7d6dd16205c59db28634458bbba294f76

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87341 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31430 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18152 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96570 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149916 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91316 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29793 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79458 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91343 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92958 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/24071 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74133 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23913 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79042 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79265 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66929 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79193 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27509 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13100 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72874 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27461 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14115 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25951 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/2726 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29148 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/36977 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75694 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29079 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33392 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16766 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -22,7 +22,7 @@
 
 #include "BaseTarget.xcconfig"
-#include "WebKitTargetConditionals.xcconfig"
+#include "../../../Configurations/WebKitTargetConditionals.xcconfig"
```
It's already included from BaseTarget.xcconfig, so this line is unnecessary.

ddkilzer replied:
True for all three of these, so I'll made that change in all of them:

```
Source/WebKit/Configurations/WebKit.xcconfig
Source/WebKit/Configurations/WebKitPlatform.xcconfig
Source/WebKit/Configurations/WebKitSwift.xcconfig
```

Commit:
Unify copies of WebKitTargetConditionals.xcconfig
https://bugs.webkit.org/show_bug.cgi?id=244440
<rdar://99233826>

Reviewed by Alexey Proskuryakov.

- Move latest copy of WebKitTargetConditionals.xcconfig into
  Configurations folder.
- Remove duplicate copies of WebKitTargetConditionals.xcconfig
  in each project.
- Remove WebKitTargetConditionals.xcconfig from Xcode projects.
- Change references to WebKitTargetConditionals.xcconfig in
  other xcconfig files to use a relative path to the single
  copy.

* Configurations/WebKitTargetConditionals.xcconfig: Rename from Source/WebGPU/Configurations/WebKitTargetConditionals.xcconfig.
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/AngleMetalLib.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebInspectorUI/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
- Remove import of WebKitTargetConditionals.xcconfig since it's
  already included in BaseTarget.xcconfig.
* Source/WebKit/Configurations/WebKitPlatform.xcconfig: Ditto.
* Source/WebKit/Configurations/WebKitSwift.xcconfig: Ditto.
* Source/WebKit/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/TestWebKitAPI/Configurations/WebKitTargetConditionals.xcconfig: Remove.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/253873@main
